backend/src/api/open_source_project.ts

```diff
@@ -0,0 +1,263 @@
+   };
+ }
+
+} catch (error) {
+ return {
+   statusCode: 404,
+   body: JSON.stringify(error)
+ };
+}
+});
+
+
+
+
+/**
+ * @swagger
+ *
+ * /project/grabInfo/{projectId}:
+ *  get:
+ *    description: Get information about a particular open source project.
+ *    parameters:
+ *      - in: path
+ *        name: projectId
+ *        description: Open Source Project ID
+ *    tags:
+ *    - OpenSourceProjects
+ */
+export const getById = wrapHandler(async (event) => {
+  const id = event.pathParameters?.projectId;
+  if (!id) {
+    return {
+      statusCode: 400,
+      body: JSON.stringify({ error: 'Project ID is required' }),
+    };
+  }
+ 
+  await connectToDatabase();
+
+  const project = await OpenSourceProject.findOne(id, {
+    relations: ['organizations'],
+  });
+
+  if (!project) {
+    return {
+      statusCode: 404,
+      body: JSON.stringify({ error: 'Project not found' }),
+    };
+  }
+ 
+  // retrieve associated orgid(s)
+  const orgIds = project.organizations.map(org => org.id);
+  const userOrgs = getOrgMemberships(event);
+  
+  if (!isGlobalViewAdmin(event) && !isGlobalWriteAdmin(event) && !userOrgs.some(userId => orgIds.includes(userId))) return Unauthorized;
+  
+  return {
+    
+    statusCode: 200,
+    body: JSON.stringify(project),
+  };
+});
+
+
+/**
+ * @swagger
+ *
+ * /project/listOrgs/{orgId}:
+ *  list:
+ *    description: List open source projects for given organization.
+ *    parameters:
+ *      - in: path
+ *        name: orgId
+ *        required: true
+ *        description: Organization ID
+ *    tags:
+ *    - Organizations
+ */
+export const listByOrg = wrapHandler(async (event) => {
+  console.log("im here");
+  const orgId = event.pathParameters?.orgId;
+  // check permissions
+  if (!orgId || (!isGlobalViewAdmin(event) && !isGlobalWriteAdmin(event) && !getOrgMemberships(event).includes(orgId))) return Unauthorized;
+  await connectToDatabase();
+  console.log("made it here");
+  const organization = await Organization.findOne({
+    where: { id: orgId },
+    relations: ['openSourceProjects']
+  });
+  console.log("my org = ", organization);
+  if (organization) {
+    return {
+      statusCode: 200,
+      body: JSON.stringify(organization.openSourceProjects)
+    };
+  }
+  return NotFound;
+});
+
+
+
+/**
+ * @swagger
+ *
+ * /project_upsert/org/{orgId}:
+ *  create:
+ *    description: Create a new open source project without an organization if it doesn't already exist and add it to a given organization.
+ *    tags:
+ *    - Organizations
+ */
+export const create_proj = wrapHandler(async (event) => {
+  const orgId = event.pathParameters?.orgId;
+  const parsedBody = JSON.parse(event.body ?? '{}');
+  const { url, hipcheckResults, organizations } = parsedBody;
+
+  // check permissions
+  if (!orgId || (!isGlobalWriteAdmin(event) && !getOrgMemberships(event).includes(orgId))) return Unauthorized;
+  
+  // const body = await validateBody(OpenSourceProject, event.body);
```

Comment:
Are you able to use validateBody instead of JSON.parse? If needed, you can create a separate type to represent a new project (e.g., without hipcheckresults) -- you can see here for an example: https://github.com/cablej/XFD/blob/main/backend/src/api/scans.ts#L261

Reply:
Added in validateBody for the create request. Should I also use validateBody for the other endpoints?